packages/react/src/components/Storage/FileUploader/Previewer/index.tsx

```diff
@@ -1,6 +1,59 @@
 import React from 'react';
-import { Card } from '../../../../primitives';
+import { translate } from '@aws-amplify/ui';
+import { PreviewerProps } from '../types';
+import { Button, Card, Flex, Text, View } from '../../../../primitives';
+import { UploadDropZone } from '../UploadDropZone';
+import { UploadButton } from '../UploadButton';
+import { Tracker } from '../Tracker';
 
-export function Previewer(): JSX.Element {
-  return <Card>Previewer</Card>;
+export function Previewer({
+  files,
+  inDropZone,
+  onClose,
+  onDragEnter,
+  onDragLeave,
+  onDragOver,
+  onDragStart,
+  onDrop,
+  ...rest
+}: PreviewerProps): JSX.Element {
+  const onClick = () => {
+    // start upload
+  };
+  return (
+    <Card variation="outlined" className="amplify-fileuploader__previewer">
+      <Flex className="amplify-fileuploader__previewer__body">
+        <UploadDropZone
+          inDropZone={inDropZone}
+          onDragEnter={onDragEnter}
+          onDragLeave={onDragLeave}
+          onDragOver={onDragOver}
+          onDragStart={onDragStart}
+          onDrop={onDrop}
+        >
+          <Text className="amplify-fileuploader__dropzone__text">
+            {translate('Drop files here or')}
+          </Text>
+          <UploadButton
+            {...rest}
+            className={'amplify-fileuploader__dropzone__button'}
+          />
+        </UploadDropZone>
+        <Text fontWeight="bold">{files.length} files selected</Text>
+        {files?.map((file, index) => (
+          <Tracker key={index}>{file.name}</Tracker>
+        ))}
+        <View className="amplify-fileuploader__footer">
+          <View>
+            <Button size="small" variation="primary" onClick={onClick}>
+              Upload {files.length} files
+            </Button>
+          </View>
+          <Button size="small" variation="link" onClick={onClose}>
+            Clear all
+          </Button>
```

Comment:
is `onClose` the right variable name for this? Maybe `onClear`? 

I would like to ask @calebpollman and @ioanabrooks too because we were misaligned in these event handlers in previous PRs.

Reply:
I'll change to onClear

Reply:
Fixed on next PR